x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -317,3 +327,37 @@ export const getAgentDataHandler: RequestHandler<
 function isStringArray(arr: unknown | string[]): arr is string[] {
   return Array.isArray(arr) && arr.every((p) => typeof p === 'string');
 }
+
+// Read a static file generated at build time
+export const getAvailableVersionsHandler: RequestHandler = async (context, request, response) => {
+  const versionsFile = 'build/kibana/x-pack/plugins/fleet/target/agent_versions_list.txt';
+  let versionsToDisplay: string[] = [];
+  const kibanaVersion = appContextService.getKibanaVersion();
+
+  try {
+    const file = await fs.readFile(versionsFile);
```

Comment:
Maybe could cache this and read the file only once, this is not expected to change while Kibana is running

Reply:
I implemented a caching mechanism for this file. Can you take a look again?